### ISSUE TITLE
class requirement: metrics output

### ISSUE THREAD

**Issue by cbaakman** [DeepRank/deeprank-gnn-2#17]
During training/testing, the GNN generates output: predictions that can be compared to ground truth values. Either by regression or classification.

To assess the quality of the output, we compute metrics. Examples: TPR, accuracy, MCC, cross entropy
There's also the question on how to store such metrics: tensorboard, csv

What I suggest is that we create a metrics output class. (we can name it later) Possibly we can make it overridable, but the idea is that we plug one or more instances of this class into the training/validation/testing functions so that they can process the GNN's output.

The class should at least have a method that is called everytime that the GNN produces new output. It should take the batch and epoch number as input, together with the GNN's output and ground truth value.

**Comment by cbaakman** [comment]
This was done in commit 969f0ffb31c031e1882e34e61a1260980fd235be.
